falcon/routing/converters.py

```diff
@@ -98,15 +98,52 @@ def convert(self, value):
         except ValueError:
             return None
 
+        return self._validate_min_max_value(value)
+
+    def _validate_min_max_value(self, value):
         if self._min is not None and value < self._min:
             return None
-
         if self._max is not None and value > self._max:
             return None
 
         return value
 
 
+class FloatConverter(IntConverter):
+    """Converts a field value to an float.
+
+    Identifier: `float`
+    Keyword Args:
+        min (float): Reject the value if it is less than this number.
+        max (float): Reject the value if it is greater than this number.
+        allow_nan (bool) : Allow the NaN value (default: False).
+    """
+
+    __slots__ = '_allow_nan'
+
+    def __init__(self, min=None, max=None, allow_nan=False):
+        self._min = min
+        self._max = max
+        self._allow_nan = allow_nan
+
+    def convert(self, value):
+        if value.strip() != value:
+            return None
+
+        try:
+            value = float(value)
+            if not self._allow_nan:
+                if self._is_nan(value):
+                    return None
+        except ValueError:
+            return None
+
+        return self._validate_min_max_value(value)
+
+    def _is_nan(self, value):
```

Comment:
I think it actually was a misunderstanding overall wrt my initial suggestion, maybe partially because naming is hard...
IMHO it would make sense to have a single parameter governing whether we allow other than finite ("ordinary") numbers or not. So basically `nan`, `inf`, `-inf`.

Reply:
makes sense. so actually use `math.isfinite(value)`

Reply:
The inequality trick executes a tiny bit faster vs invoking `math.isnan()`. But that's a moot point if we follow @vytas7's suggestion. I can push a commit to make that change.

Reply:
done

Reply:
let me know what you think